server/src/test/java/com/defold/extender/services/DefoldSDKServiceTest.java

```diff
@@ -28,50 +28,111 @@
 import static org.junit.jupiter.api.Assertions.assertFalse;
 import static org.junit.jupiter.api.Assertions.assertNotNull;
 import static org.junit.jupiter.api.Assertions.assertThrows;
 import static org.junit.jupiter.api.Assertions.assertTrue;
-import static org.mockito.Mockito.mock;
+
+import static com.github.tomakehurst.wiremock.client.WireMock.*;
 
 public class DefoldSDKServiceTest {
-    private static final String folderPath = "/tmp/defoldsdk";
-    private static final String sdkUrls[] = {
-        "http://d.defold.com/archive/stable/%s/engine/defoldsdk.zip",
-        "http://d.defold.com/archive/%s/engine/defoldsdk.zip"
-    };
-    private static final String mappingsUrls[] = {
-        "http://d.defold.com/archive/stable/%s/engine/platform.sdks.json",
-        "http://d.defold.com/archive/%s/engine/platform.sdks.json"
-    };
-    private static final int sdkCacheSize = 3;
-    private static final int mappingsCacheSize = 3;
+    private static DefoldSdkServiceConfiguration configuration;
+    private static DefoldSdkServiceConfiguration zeroCacheConfiguration;
+    private static DefoldSdkServiceConfiguration otherLocationConfiguration;
+
+    private static WireMockServer mockServer;
+    private static int serverPort = 8090;
+    private static Path tmpHTTPRoot = Path.of("/tmp/__defoldsdk_http");
 
     @BeforeAll
     public static void beforeAll() throws IOException {
-        Files.createDirectories(Path.of(folderPath));
+        DefoldSDKServiceTest.configuration = DefoldSdkServiceConfiguration.builder()
+            .location(Path.of("/tmp/defoldsdk"))
+            .sdkUrls(new String[]{"http://d.defold.com/archive/stable/%s/engine/defoldsdk.zip", "http://d.defold.com/archive/%s/engine/defoldsdk.zip"})
+            .mappingsUrls(new String[] {"http://d.defold.com/archive/stable/%s/engine/platform.sdks.json", "http://d.defold.com/archive/%s/engine/platform.sdks.json"})
+            .cacheSize(3)
+            .mappingsCacheSize(3)
+            .cacheClearOnExit(true)
+            .enableSdkVerification(false)
+            .maxVerificationRetryCount(3)
+            .build();
+
+        DefoldSDKServiceTest.zeroCacheConfiguration = new DefoldSdkServiceConfiguration(DefoldSDKServiceTest.configuration.toBuilder());
+            zeroCacheConfiguration.setCacheSize(0);
+    
+        Files.createDirectories(DefoldSDKServiceTest.configuration.getLocation());
+
+        DefoldSDKServiceTest.otherLocationConfiguration = new DefoldSdkServiceConfiguration(DefoldSDKServiceTest.zeroCacheConfiguration.toBuilder());
+        DefoldSDKServiceTest.otherLocationConfiguration.setLocation(Path.of("/tmp/defoldsdk_test"));
+        Files.createDirectories(DefoldSDKServiceTest.otherLocationConfiguration.getLocation());
+
+        // prepare content for serving
+        Files.createDirectories(tmpHTTPRoot);
+        FileUtils.copyDirectory(new File("test-data/checksum_sdk/"), new File(tmpHTTPRoot.toFile(), "__files"), File::isFile);
```

Comment:
Copied files to temporary folder because WireMock serves files only from `__files` folder.